apps/meteor/server/methods/sendWelcomeEmail.ts

```diff
@@ -0,0 +1,49 @@
+import { Users } from '@rocket.chat/models';
+import { Meteor } from 'meteor/meteor';
+
+import * as Mailer from '../../app/mailer/server/api';
+import { settings } from '../../app/settings/server';
+import { isSMTPConfigured } from '../../app/utils/server/functions/isSMTPConfigured';
+
+export async function sendWelcomeEmail(to: string): Promise<void> {
+	if (typeof to !== 'string') {
+		throw new Meteor.Error('error-invalid-arguments', 'Invalid arguments', {
+			method: 'sendWelcomeEmail',
+		});
+	}
+
+	if (!isSMTPConfigured()) {
+		throw new Meteor.Error('error-email-send-failed', 'SMTP is not configured', {
+			method: 'sendWelcomeEmail',
+		});
+	}
+
+	const email = to.trim();
+
+	const user = await Users.findOneByEmailAddress(email, { projection: { _id: 1 } });
+
+	if (!user) {
+		throw new Meteor.Error('error-invalid-user', 'Invalid user', {
+			method: 'sendWelcomeEmail',
+		});
+	}
+
+	try {
+		let html = '';
+		Mailer.getTemplate('Accounts_UserAddedEmail_Email', (template) => {
+			html = template;
+		});
```

Comment:
Curious: will the callback on getTemplate be called always before Mailer.send needs the data?

Reply:
Yes, it will, settings are already "ready" let's say, also, the Mailer.getTemplate does not return a Promise, so we can't (it's going to have any effect) use await here